vault/resource_database_secret_backend_connection.go

```diff
@@ -0,0 +1,613 @@
+			},
+			"verify_connection": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Specifies if the connection is verified during initial configuration.",
+				Default:     true,
+			},
+			"allowed_roles": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "A list of roles that are allowed to use this connection.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"cassandra": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Connection parameters for the cassandra-database-plugin plugin.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"hosts": {
+							Type: schema.TypeList,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Optional:    true,
+							Description: "Cassandra hosts to connect to.",
+						},
+						"port": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Description: "The transport port to use to connect to Cassandra.",
+							Default:     9042,
+						},
+						"username": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The username to use when authenticating with Cassandra.",
+						},
+						"password": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The password to use when authenticating with Cassandra.",
+							Sensitive:   true,
+						},
+						"tls": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Whether to use TLS when connecting to Cassandra.",
+							Default:     true,
+						},
+						"insecure_tls": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Whether to skip verification of the server certificate when using TLS.",
+							Default:     false,
+						},
+						"pem_bundle": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Concatenated PEM blocks containing a certificate and private key; a certificate, private key, and issuing CA certificate; or just a CA certificate.",
+							Sensitive:   true,
+						},
+						"pem_json": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Specifies JSON containing a certificate and private key; a certificate, private key, and issuing CA certificate; or just a CA certificate.",
+							Sensitive:   true,
+						},
+						"protocol_version": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     2,
+							Description: "The CQL protocol version to use.",
+						},
+						"connect_timeout": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     5,
+							Description: "The number of seconds to use as a connection timeout.",
+						},
+					},
+				},
+				MaxItems:      1,
+				ConflictsWith: []string{"mongodb", "hana", "mssql", "mysql", "postgresql", "oracle"},
```

Comment:
I feel like this list repetition is pretty error prone, perhaps you could write a little func that just takes the removal in?